src/Language/PureScript/Ide/Filter.hs

```diff
@@ -15,112 +15,130 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 
 module Language.PureScript.Ide.Filter
-       ( Filter
-       , declarationTypeFilter
-       , namespaceFilter
+       ( Filter(..)
```

Comment:
Unless I'm missing something I think you can continue not to export the constructor?